website/docs/feature-flag-tutorials/use-cases/manage-feature-flags-in-code.mdx

```diff
@@ -0,0 +1,385 @@
+</TabItem>
+</Tabs>
+
+## Minimizing tech debt and managing the flag lifecycle
+
+Let's face it: stale flags are [tech debt](/reference/technical-debt). Without a plan, your codebase will fill up with forgotten and risky flags. The only way to win is with a clear process for managing their entire [lifecycle](/reference/feature-toggles#feature-flag-lifecycle).
+
+The first line of defense is clarity. A flag named `temp_fix_v2` is a mystery waiting to happen.
+Enforce a strict [naming convention](/reference/feature-toggles#set-a-naming-pattern) that encodes key information. A highly effective pattern is: `[team]_[feature-name]_[issue-number]`. For example: `checkout_multistep-payment-flow_jira-376`.
+
+You can also use code comments to share additional information and use the description and [external links](/reference/feature-toggles#external-links) features in Unleash to tie the flag back to the work that created it.
+
+### Flag cleanup best practices
+
+You can use a flag's [lifecycle data](/reference/feature-toggles#feature-flag-lifecycle) and automated reminders to ensure flags are removed from your code once they have served their purpose.
+
+Here's our recommended workflow for flag cleanup:
+
+- **Update the flag's lifecycle status**: Once a feature is stable and fully rolled out, mark it as *Completed* in Unleash. If you forget, Unleash will prompt you to update the status for stale flags that are no longer sending metrics. This moves the flag to the Cleanup lifecycle stage, creating a clear backlog for removal.
+
+- **Clean up the code**: Remove the conditional logic, any old code paths, the flag's definition from your central file, and any helper methods from your wrapper. This ensures you remove dead code and reduce complexity.
+
+- **Test and deploy**: Run your tests to ensure everything still works as expected, then deploy your changes.
+
+- **Archive the flag in Unleash**: Finally, archive the flag in the Unleash UI. Don't delete it—archiving preserves its history for auditing and analysis.
+
+Just hoping that people remember to clean up is not a sustainable strategy. You need to automate your governance process.
+
+Here are some practical tips:
+- **Automated ticketing**: Use webhooks or integrations to automatically create "Remove Flag" tickets.
+- **Scheduled reviews**: Make flag reviews a part of your process, for example your planning. Teams should justify a flag's existence or schedule them for removal.
+- **Update "Definition of Done"**: A feature isn't "done" until its associated feature flag has been removed from the code and archived in Unleash.
+- **Use AI to speed up the cleanup**: Rely on [AI coding assistants](https://www.getunleash.io/blog/ai-flag-cleanup) to automate and fixing flag removal issues.
+
+
+## Testing with feature flags
+
+A common fear is that flags will cause a "combinatorial explosion" of test cases. Don't worry, you don't need to test every possible combination. Instead, focus on these key scenarios:
```

Comment:
Every company will have a different testing strategy so I wouldn't recommend one approach.
Some useful testing strategies:
* all flags on in smoke or e2e tests (whole application) and component (one service) tests
* all flags off in smoke or e2e tests (whole application) and component (one service) tests
* most common combination of flags in smoke or e2e tests (whole application) and component (one service) tests
* in unit tests flag on and off for one flag that is affecting the unit

Reply:
updated in 7ef76fc